packages/codecs-core/src/resize-codec.ts

```diff
@@ -0,0 +1,103 @@
+import { SOLANA_ERROR__CODECS__EXPECTED_POSITIVE_BYTE_LENGTH, SolanaError } from '@solana/errors';
+
+import {
+    Codec,
+    createDecoder,
+    createEncoder,
+    Decoder,
+    Encoder,
+    FixedSizeCodec,
+    FixedSizeDecoder,
+    FixedSizeEncoder,
+    isFixedSize,
+} from './codec';
+import { combineCodec } from './combine-codec';
+
+/**
+ * Updates the size of a given encoder.
+ */
+export function resizeEncoder<TFrom, TSize extends number, TNewSize extends number>(
+    encoder: FixedSizeEncoder<TFrom, TSize>,
+    resize: (size: TSize) => TNewSize,
+): FixedSizeEncoder<TFrom, TNewSize>;
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
```

Comment:
Damn.

I was able to get this to work without `any` by using `Encoder<never>`, `Decoder<unknown>` and `Codec<never, unknown>`, but then `Codec<never, unknown>` itself becomes a type error. Can't decide which is worse: this any suppression, or…

```ts
export function resizeCodec<
    TCodec extends Codec<
        never,
        // eslint-disable-next-line @typescript-eslint/ban-ts-comment
        // @ts-expect-error
        unknown
    >,
>(codec: TCodec, resize: (size: number) => number): TCodec;
```

Reply:
Whilst `any` is a bad word in TypeScript, I do think it describes best the situation here — i.e. we accepts codecs for _any_ types and return you the exact same type.